graphql/task_resolver.go

```diff
@@ -579,6 +579,25 @@ func (r *taskResolver) TaskLogs(ctx context.Context, obj *restModel.APITask) (*T
 
 // Tests is the resolver for the tests field.
 func (r *taskResolver) Tests(ctx context.Context, obj *restModel.APITask, opts *TestFilterOptions) (*TaskTestResult, error) {
+	if opts != nil && len(opts.Statuses) > 0 {
+		queryingFailingTestsOnly := true
+		if len(opts.Statuses) <= len(evergreen.TestFailureStatuses) {
+			for _, status := range opts.Statuses {
+				if !evergreen.IsFailedTestStatus(status) {
+					queryingFailingTestsOnly = false
+					break
+				}
+			}
+		}
```

Comment:
I think there are easier ways to check this condition using utility functions such as `utility.FilterSlice` or `utility.GetSetDifference`